openapi_python_client/templates/property_templates/union_property.py.jinja

```diff
@@ -1,19 +1,31 @@
 {% macro construct(property, source, initial_value=None) %}
-def _parse_{{ property.python_name }}(data: Any) -> {{ property.get_type_string() }}:
-    data = None if isinstance(data, Unset) else data
-    {{ property.python_name }}: {{ property.get_type_string() }}
+def _parse_{{ property.python_name }}(data: object) -> {{ property.get_type_string() }}:
+    {% if "None" in property.get_type_strings_in_union(json=True) %}
+    if data is None:
+        return data
+    {% endif %}
+    {% if "Unset" in property.get_type_strings_in_union(json=True) %}
+    if isinstance(data, Unset):
+        return data
+    {% endif %}
```

Comment:
I think we can use `property.required` and `property.nullable` for this? Maybe that's not robust enough, not sure.

Reply:
I think that depends on the unrolling unions thing you had mentioned below, since a nested property could still be nullable. And technically I guess you could have a none-type in the union.